Source/Urho3D/CSharp/Managed/Scene/Node.cs

```diff
@@ -34,9 +38,68 @@ public partial class Node
             return (T)GetComponent(typeof(T).Name, recursive);
         }
 
+        /// <summary>
+        /// Get first occurance of a component type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
         public T GetComponent<T>() where T : Component
         {
             return (T)GetComponent(typeof(T).Name);
         }
+
+        /// <summary>
+        /// get all components of a type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public ComponentList GetComponentsbyType<T>(bool recursive = false) where T : Component
+        {
+            ComponentList cplist = new ComponentList();
+            GetComponents(cplist, typeof(T).Name, recursive);
+            return cplist;
+        }
+
+        /// <summary>
+        /// Remove all Components of a specific Type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void RemoveAllComponents<T>() where T : Component
+        {
+            var childrenNode = GetComponentsbyType<T>();
+            foreach (var childnode in childrenNode)
+            {
+                RemoveComponent(childnode);
+            }
+            Debug.Assert(GetComponent<T>() is null, "Components are not removed");
+        }
+
+        /// <summary>
+        /// Remove all childern of a specific Type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void RemoveAllChildrenbyName(string NodeName, bool recursive = false)
+        {
+            var childrenNode = GetChildren(recursive).Where(o => o.Name == NodeName);
+            foreach (var childnode in childrenNode)
+            {
+                RemoveChild(childnode);
+            }
+            Debug.Assert(GetChild(NodeName) is null, "Children are not removed");
```

Comment:
Same thing about assert. Besides `GetChildren()` is recursive and `GetChild(NodeName)` is not recursive so assert tries to validate something else than what was just performed.

LINQ is slow and we absolutely do not want to use it in API.

I am not sure we should add this method at all. C++ does API does not have such method and it really boils down to two lines of user code:
```cs
foreach (var childnode in Node.GetChildren(recursive).Where(o => o.Name == NodeName))
    Node.RemoveChild(childnode);
```

Reply:
well, my understanding `foreach (var childnode in`  **Node.GetChildren(recursive).Where(o => o.Name == NodeName)** )
will linq on every round, that's why i placed it in `childrenNode` first outside the loop. !!

Reply:
not required... I will omit this one as well...